builder/dockerfile/builder.go

```diff
@@ -187,7 +187,7 @@ func buildLabelOptions(labels map[string]string, stages []instructions.Stage) {
 func (b *Builder) build(ctx context.Context, source builder.Source, dockerfile *parser.Result) (*builder.Result, error) {
 	defer b.imageSources.Unmount()
 
-	stages, metaArgs, err := instructions.Parse(dockerfile.AST)
+	stages, metaArgs, err := instructions.Parse(dockerfile.AST, nil)
```

Comment:
Would the "correct" way be to pass a linter with `SkipAll`? `linter.New(linter.Config{SkipAll: true})`, basically a "dummy" linter?

Reply:
I think both are "correct" from BK perspective - I like the `nil` though as it saves an unnecessary allocation.

Reply:
Let me know if you prefer the `linter.New` though 👍🏻

Reply:
It's not a blocker for sure! 

I'm fine either way; I'm ok with the `nil` - and this is still something we can change later. I was mostly pondering "what's correct here"

Also (honestly) I was hoping that linter would be an _interface_ here, but it looks to be a _type_. If it was an interface, then (possibly?) it would make it easier to swap implementations, and we could even have a linter for the classic builder to assist with deprecations, as well as a better way to indicate "this feature is not supported by the classic builder").


cc @daghack @jsternberg